[Bug]: Dev docker-compose startup fails on fresh clone, missing shared-schemas and ui build steps

### What's the bug?

Following the contributor setup in CONTRIBUTING.md (`docker compose up`) on a fresh clone crashes the `insforge` container with exit code 1.

The dev `docker-compose.yml` startup command doesn't build `shared-schemas` or `ui` before starting the dev server. Since their `dist/` directories are gitignored, they don't exist on a fresh clone — causing the frontend to crash during Vite's dependency scan.

The README's self-hosted setup uses `docker-compose.prod.yml` (which builds everything via the multi-stage Dockerfile), so end-user setup isn't affected — only the contributor dev workflow from CONTRIBUTING.md.

**Error from logs:**
```
[frontend] ✘ [ERROR] Could not resolve "./schedules.schema"
    ../shared-schemas/src/index.ts:23:14
```

**Why others may not have hit this:**
If a developer runs `npm install && npm run build` locally before Docker, the `dist/` directories exist on the host. Docker bind-mounts them into the container, and everything works. Only a pure Docker-first setup (fresh clone → `docker compose up`) triggers the crash.

**Proposed fix:**
Add build steps to the startup command in `docker-compose.yml`:
```yaml
# Before:
command: sh -c "npm install && cd backend && npm run migrate:up && cd .. && npm run dev"

# After:
command: sh -c "npm install && npm run build:shared-schemas && npm run build:ui && cd backend && npm run migrate:up && cd .. && npm run dev"
```

This matches the production Dockerfile (Stage 3) which already runs `npm run build` before starting.

I'd like to work on this fix.

### How to reproduce

1. `git clone https://github.com/InsForge/InsForge.git && cd InsForge`
2. `cp .env.example .env`
3. `docker compose up` (as instructed in CONTRIBUTING.md line 60)
4. Container `insforge` exits with code 1
5. `docker compose logs insforge` shows the Vite dependency scan error

### Environment (optional)

- OS: Windows 11 Home 10.0.26200
- Docker Desktop (Docker Compose v2)
- Fresh clone, no prior local `npm install` or `npm run build`

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -148,7 +148,7 @@ services:
       - ui_node_modules:/app/ui/node_modules
       - shared-logs:/insforge-logs
       - storage-data:/insforge-storage
-    command: sh -c "npm install && cd backend && npm run migrate:up && cd .. && npm run dev"
+    command: sh -c "npm install && npm run build:shared-schemas && npm run build:ui && cd backend && npm run migrate:up && cd .. && npm run dev"
     restart: unless-stopped
     security_opt:
       - no-new-privileges:true

